LM IR BLinker.X/main.c

```diff
@@ -43,9 +43,21 @@
 
 #include "mcc_generated_files/mcc.h"
 
-/*
-                         Main application
- */
+// ========== DEFINES ==========
+#define LED_PIN     LATAbits.LATA3
+#define PWM_EN      PWM3CONbits.EN
+ 
+
+// ========== GLOBAL VARIABLES ==========
+const bool pattern[] = {0, 1, 0, 1, 1, 1};
+uint8_t patternIndex = 0;
+
+
+// ========== FUNCTION PROTOTYPES ==========
+void bitTimerInterrupt();

```

Comment:
If you're wondering what a prototype is, it's basically a function declaration.

In C, there are declarations and definitions. Definitions are where you say what a function does, or what a variable is. It creates an object which gets linked together at the end. Declarations (which are not required for a definition) are telling the compiler that such a definition is **going** to happen at some point... we promise.

I like sticking non-important function definitions below my main. But in order to do that, the compiler needs to know such a function exists before you reference it (like I"m about to do in main()). Therefore: prototype (function declaration)

Reply:
Also, along those lines of declarations vs. definitions: you may notice that in your project structure, there are *.h files that contain **prototypes**/**declarations** of things, then those things have **definitions** in the *.c files. That is the conventional way of splitting up files in C. Then in a file that is going to use some of that code, just `#include "*.h"` to use those objects.

Reply:
There is definitely some learnedness happening here. And I appreciate the detailed explanations of what you are doing, now time for me to dig into the code and see if I can understand it haha.